core/src/main/scala/com/salesforce/op/stages/impl/tuning/DataBalancer.scala

```diff
@@ -116,26 +116,39 @@ class DataBalancer(uid: String = UID[DataBalancer]) extends Splitter(uid = uid)
     }
   }
 
+
   /**
-   * Split into a training set and a test set and balance the training set
+   * Function to set parameters before passing into the validation step
+   * eg - do data balancing or dropping based on the labels
    *
-   * @param data to prepare for model training. first column must be the label as a double
-   * @return balanced training set and a test set
+   * @param data
+   * @return Parameters set in examining data
    */
-  def prepare(data: Dataset[Row]): ModelData = {
+  override def preValidationPrepare(data: Dataset[Row]): Option[SplitterSummary] = {
     val negativeData = data.filter(_.getDouble(0) == 0.0).persist()
     val positiveData = data.filter(_.getDouble(0) == 1.0).persist()
     val negativeCount = negativeData.count()
     val positiveCount = positiveData.count()
     val seed = getSeed
 
-    if (!(isSet(isPositiveSmall) ||
-      isSet(downSampleFraction) ||
-      isSet(upSampleFraction) ||
-      isSet(alreadyBalancedFraction))
-    ) {
-      estimate(positiveCount = positiveCount, negativeCount = negativeCount, seed = seed)
-    }
+    estimate(positiveCount = positiveCount, negativeCount = negativeCount, seed = seed)
+
+    summary
+  }
+
+  /**
+   * Rebalance the training data within the validation step
+   *
+   * @param data to prepare for model training. first column must be the label as a double
+   * @return balanced training set and a test set
+   */
+  def validationPrepare(data: Dataset[Row]): Dataset[Row] = {
+
+    if (summary.isEmpty) throw new RuntimeException("Cannot call prepare until examine has been called")
```

Comment:
`examine` was renamed to `preValidationPrepare`? so please update the error message.